### PR TITLE
Sp/tuna mituna interface typehint

### PR DESCRIPTION
**Pull request by technicalgrp89**
tuna/mituna_interface.py
- type hints support
- mypy cmd support - utils.groovy update

### CODE REVIEW

**Comment by alexandraBara** [comment]
Please merge with develop

File: tuna/mituna_interface.py [hunk 1]
```diff
@@ -177,10 +190,11 @@ def get_kwargs(self, gpu_idx, f_vals):
         'job_queue_lock': f_vals["job_queue_lock"],
         'result_queue': f_vals["result_queue"],
         'result_queue_lock': f_vals["result_queue_lock"],
-        'label': self.args.label,
-        'docker_name': self.args.docker_name,
+        'label': self.args.label,  #type: ignore
+        'docker_name': self.args.docker_name,  #type: ignore
```
Comment (alexandraBara) [inline tuna/mituna_interface.py]: why would you ignore the type here? they are both strings

File: tuna/mituna_interface.py [hunk 2]
```diff
@@ -177,10 +190,11 @@ def get_kwargs(self, gpu_idx, f_vals):
...
         'end_jobs': f_vals['end_jobs'],
-        'session_id': self.args.session_id
+        'session_id':
+            self.args.session_id  #type: ignore
```
Comment (alexandraBara) [inline tuna/mituna_interface.py]: this would be an int

**Review by alexandraBara — CHANGES_REQUESTED** [review]

File: tuna/mituna_interface.py (hunk 2 above)
Comment (technicalgrp89) [reply tuna/mituna_interface.py]: logic changed

File: tuna/mituna_interface.py (hunk 1 above)
Comment (technicalgrp89) [reply tuna/mituna_interface.py]: logic changed

File: tuna/mituna_interface.py [hunk 3]
```diff
@@ -35,18 +35,21 @@
 from tuna.libraries import Library
 from tuna.utils.logger import setup_logger
 from tuna.utils.utility import get_env_vars
+from tests.utils import GoFishArgs
+from tests.utils import ExampleArgs
```
Comment (alexandraBara) [inline tuna/mituna_interface.py]: these are strictly defined for testing purposed, please do not use in /tuna

File: tuna/mituna_interface.py [hunk 4]
```diff
@@ -35,18 +35,21 @@
...
   common functionalities. """
 
   def __init__(self, library=Library.MIOPEN) -> None:
+
     self.library: Library = library
 
     self.logger: logging.Logger = setup_logger(logger_name=self.library.value,
                                                add_streamhandler=True)
-    self.args = None
+    self.args: Union[GoFishArgs, ExampleArgs] = None
```
Comment (alexandraBara) [inline tuna/mituna_interface.py]: it should be a Namespace not a Union of these classes used in /tests

File: tuna/mituna_interface.py [hunk 5]
```diff
@@ -192,15 +195,10 @@ def get_kwargs(self, gpu_idx: int, f_vals: Dict[str, Any]) -> Dict[str, Any]:
...
+        'session_id': self.args.session_id
     }
-    if self.args is not None:
-      kwargs['label'] = self.args.label
-
-    if self.args is not None:
-      kwargs['docker_name'] = self.args.docker_name
-
-    if self.args is not None:
-      kwargs['session_id'] = self.args.session_id
```
Comment (alexandraBara) [inline tuna/mituna_interface.py]: If you cant get around the None check, please do:
if self.args.label is not None:
  kwargs['label'] = self.args.label
same for docker_name and session_id

**Review by alexandraBara — CHANGES_REQUESTED** [review]

**Comment by technicalgrp89** [comment]
@alexandraBara @JehandadKhan

File: tuna/mituna_interface.py [hunk 6]
```diff
@@ -56,21 +66,23 @@ def check_docker(self, worker, dockername="miopentuna"):
...
       for line in out.readlines():
-        if line.find(dockername) != -1:
-          self.logger.warning('%s docker image exists', dockername)
-          break
+        if line is not None:
+          if line.find(dockername) != -1:
+            self.logger.warning('%s docker image exists', dockername)
+            break
       if line is None:
         self.logger.warning('%s docker image does not exist', dockername)
```
Comment (alexandraBara) [inline tuna/mituna_interface.py:78]:
```suggestion
        if line is not None:
          if line.find(dockername) != -1:
            self.logger.warning('%s docker image exists', dockername)
            break
      else:
        self.logger.warning('%s docker image does not exist', dockername)
```

**Review by alexandraBara — CHANGES_REQUESTED** [review]
Looks good, just the one small change

**Review by alexandraBara — APPROVED** [review]

**Review by JehandadKhan — APPROVED** [review]

**Pull request merged**